pyccel/codegen/printing/fcode.py

```diff
@@ -231,11 +231,11 @@
     'C_BOOL'                : 'b1'
 }
 
-inc_keyword = (r'do\b', r'if\b',
+inc_keyword = (r'do\b', r'if \(.*?\) then$',
                r'else\b', r'type\b\s*[^\(]',
                r'(elemental )?(pure )?(recursive )?((subroutine)|(function))\b',
                r'interface\b',r'module\b(?! *procedure)',r'program\b')
-inc_regex = re.compile('|'.join('({})'.format(i) for i in inc_keyword))
+inc_regex = re.compile('|'.join(f'({i})' for i in inc_keyword))
```

Comment:
Could you please explain the change in `inc_keyword`?

Reply:
Previously we indented after a line starting with the word if. This means that we would indent after:
```fortran
if (allocated(arr)) deallocate(arr)
```
because it starts with `if`. The new regex means we indent after a line starting with the word if and ending with the word `then`